domain/crossmodelrelation/state/model/relation_network.go

```diff
@@ -173,25 +174,40 @@ func (st *State) NamespacesForRelationEgressNetworksWatcher() (string, string, s
 
 // InitialWatchStatementForRelationEgressNetworks returns the initial query
 // for watching relation egress networks. It returns the actual egress CIDRs
-// if the relation exists, or an empty slice if no egress networks are
-// configured.
+// following the same priority logic as the watcher mapper:
+// 1. Relation-specific egress CIDRs from relation_network_egress
+// 2. Model config egress-subnets
+// 3. Unit public addresses converted to CIDRs
 func (st *State) InitialWatchStatementForRelationEgressNetworks(relationUUID string) eventsource.NamespaceQuery {
 	return func(ctx context.Context, runner database.TxnRunner) ([]string, error) {
 		db, err := st.DB(ctx)
 		if err != nil {
 			return nil, errors.Capture(err)
 		}
 
-		checkStmt, err := st.Prepare(`
-SELECT &uuid.*
-FROM   relation
-WHERE  uuid = $uuid.uuid
-`, uuid{})
+		// Step 1: Get unit addresses for this relation
+		// If there are no units, return empty regardless of configured CIDRs
+		unitAddressStmt, err := st.Prepare(`
+SELECT u.uuid AS &unitAddress.unit_uuid,
+       ua.address_value AS &unitAddress.address_value,
+       ua.config_type_name AS &unitAddress.config_type_name,
+       ua.type_name AS &unitAddress.type_name,
+       ua.origin_name AS &unitAddress.origin_name,
+       ua.scope_name AS &unitAddress.scope_name,
+       ua.space_uuid AS &unitAddress.space_uuid,
+       ua.cidr AS &unitAddress.cidr
+FROM   relation_unit ru
+JOIN   relation_endpoint re ON ru.relation_endpoint_uuid = re.uuid
+JOIN   unit u ON ru.unit_uuid = u.uuid
```

Comment:
todo: use explicit aliasing per the sql best practices.

Reply:
Done.